helion/language/tunable_ops.py

```diff
@@ -220,3 +240,90 @@ def _register_tunable_codegen(state: CodegenState) -> ast.AST:
     config_value = state.config[name]
     assert isinstance(config_value, (int, float, bool))
     return expr_from_string(constant_repr(config_value))
+
+
+@_decorators.ref(register_tunable)
+def _(name: str, fragment: ConfigSpecFragment) -> int:
+    """Reference implementation of register_tunable."""
+    default_value = fragment.default()
+    # Convert to int if it's not already
+    if isinstance(default_value, bool):
+        return int(default_value)
+    if isinstance(default_value, int):
+        return default_value
+    # For other types (like float), convert to int
+    return int(default_value)  # type: ignore[arg-type]
+
+
+@_decorators.api(is_device_only=False)
+def next_power_of_2(n: int) -> int:
+    """
+    Return the smallest power of 2 greater than or equal to n.
+
+    This function is useful for determining allocation sizes that are powers of 2,
+    which can be more efficient for GPU memory access patterns.
+
+    Args:
+        n: A positive integer
+
+    Returns:
+        int: The smallest power of 2 >= n
+    """
+    raise NotInsideKernel
+
+
+@_decorators.ref(next_power_of_2)
+def _(n: int) -> int:
+    """Reference implementation - in ref mode, return n unchanged."""
+    return n
```

Comment:
This seems wrong... do we really want to change the semantics of next_power_of_2 in ref mode?

Reply:
fixed the issue to support next_power_of_2 - the tricky part was to support `acc += x[tile, :] + 1` where the dim sizes of `acc` and `x` don't align due to next_power_of_2 usage when creating `acc`: https://github.com/pytorch-labs/helion/blob/f5594ca5dc6c5254f6d875d962048e8e0da6832b/test/test_specialize.py#L96-L97

Now fixed by creating `RefTensor` tensor subclass that tracks both logical and physical sizes of the tensor and apply mask appropriately.